### PR TITLE
Carousel

### PR DESCRIPTION
**Pull request by utkarsh3326**
..

### CODE REVIEW

File: examples/App.js [hunk 1]
```diff
@@ -6,313 +6,14 @@
...
-              marginRight: 140,
-            },
-            titleStyle: {color: 'white'},
-          },
-        },
-      ],
-    },
-  ],
-};
+import {source} from './sampleJSON/carouselJson';
```
Comment (mahima-vs) [inline examples/App.js:13]: We should use same file for all the jsons, please put your json in jsondata.js file and use for there. then, you don't need to change the import, just need to choose your json object from source.

File: examples/src/components/header.js [hunk 2]
```diff
@@ -25,8 +27,16 @@ export default class HeaderComponent extends React.Component {
       });
     }
   }
-
+  static contextType = NavigationContext;
+  componentDidMount() {
+    if (this.props.onPress) {
```
Comment (mahima-vs) [inline examples/src/components/header.js:32]: This code can be written as-  
if (this.props.onPress &&  this.props.onPress.navigation) {
   this.props.createScreen(this.props.onPress)
    }

File: examples/src/components/header.js [hunk 3]
```diff
@@ -41,13 +51,28 @@ export default class HeaderComponent extends React.Component {
             {this.props[component].childrens.map((item, i) => {
               if (item.type == 'icon') {
                 const styleOfIcon = item.properties;
+                if (item.properties.onPress) {
```
Comment (mahima-vs) [inline examples/src/components/header.js:54]: Same as above.

File: examples/src/components/header.js (hunk 2 above)
Comment (utkarsh3326) [reply examples/src/components/header.js:32]: Yes Mam i had tried this but it was showing error don't know why , i ll check it again

File: examples/src/components/image.js [hunk 4]
```diff
@@ -39,10 +40,10 @@ ImageComponent.propTypes = {
     borderColor: PropTypes.string,
     borderRadius: PropTypes.number,
     borderWidth: PropTypes.number,
-    height: PropTypes.number,
+    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
     opacity: PropTypes.number,
     overflow: PropTypes.strig,
```
Comment (saksham695) [inline examples/src/components/image.js:45]: overflow should be PropTypes.string . There is a typo , you have wriiten PropTypes.strig

File: examples/src/components/button.js [hunk 5]
```diff
@@ -23,8 +24,15 @@ export default class ButtonComponent extends React.Component {
       properties: {},
     };
   }
+  static contextType = NavigationContext;
 
   async componentDidMount() {
+    // Creating Screen for Button onPress
+    if (this.props.onPress) {
```
Comment (mahima-vs) [inline examples/src/components/button.js:31]: if (this.props.onPress) {
      this.props.onPress.navigation

combine these 2 conditions.

File: examples/src/components/button.js [hunk 6]
```diff
@@ -63,11 +71,17 @@ export default class ButtonComponent extends React.Component {
...
   render() {
+    const navigation = this.context;
     return (
       <View>
         <Button
           title={this.state.properties.title}
-          onPress={() => this.state.properties.onPress()}
+          // onPress={() => this.state.properties.onPress()}
+          onPress={() => {
+            this.props.onPress.navigation
```
Comment (mahima-vs) [inline examples/src/components/button.js:81]: onPress condition is missing here, if incase onPress is not there in props, it will fail. Please fix.

Comment (utkarsh3326) [reply examples/src/components/button.js]: done

File: examples/src/components/button.js (hunk 5 above)
Comment (utkarsh3326) [reply examples/src/components/button.js]: done

File: examples/src/components/image.js (hunk 4 above)
Comment (utkarsh3326) [reply examples/src/components/image.js]: done

File: examples/src/components/header.js (hunk 3 above)
Comment (utkarsh3326) [reply examples/src/components/header.js:54]: done

File: examples/src/components/header.js (hunk 2 above)
Comment (utkarsh3326) [reply examples/src/components/header.js:32]: done

File: examples/App.js (hunk 1 above)
Comment (utkarsh3326) [reply examples/App.js]: done

File: examples/src/components/carousel.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import Carousel, {Pagination} from 'react-native-snap-carousel';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class MyCarousel extends React.Component {
```
Comment (mahima-vs) [inline examples/src/components/carousel.js:5]: MyCarousel doesn't seems a good name, pls change it.

File: examples/src/components/carousel.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  constructor(props) {
+    super(props);
+    this._renderItem = this._renderItem.bind(this);
+  }
+
+  _renderItem({item}) {
+    return <>{item}</>;
+  }
+
+  get pagination() {
```
Comment (mahima-vs) [inline examples/src/components/carousel.js:19]: Can we write a description about what does this method does ?

File: index.js [hunk 9]
```diff
@@ -36,11 +39,46 @@ export default class UiBuilder extends React.Component {
...
-        return <ButtonComponent {...component.properties} key={index} />;
+        return (
+          <ButtonComponent
+            {...component.properties}
+            createScreen={this.props.createScreen}
+            key={index}
+          />
+        );
+      case "carousel": {
+        let tempData = component.properties.data.map((componentData, i) => {
```
Comment (mahima-vs) [inline index.js:50]: pls rename ```tempData ``` to "itemsData``` as here we are getting carousel items.

File: examples/src/components/carousel.js (hunk 7 above)
Comment (utkarsh3326) [reply examples/src/components/carousel.js:5]: done

File: examples/src/components/carousel.js (hunk 8 above)
Comment (utkarsh3326) [reply examples/src/components/carousel.js:19]: done

File: index.js (hunk 9 above)
Comment (utkarsh3326) [reply index.js:50]: done

File: examples/src/components/carousel.js (hunk 8 above)
Comment (mahima-vs) [reply examples/src/components/carousel.js:21]: Please re-write it as ```// Enabling pagination in the carousel```

**Pull request merged**